api/src/context.rs

```diff
@@ -65,6 +65,7 @@ pub struct Context {
     pub node_config: NodeConfig,
     gas_schedule_cache: Arc<RwLock<GasScheduleCache>>,
     gas_estimation_cache: Arc<RwLock<GasEstimationCache>>,
+    gas_limit_cache: Arc<RwLock<GasLimitCache>>,
```

Comment:
nit: FWIW, it might make sense to create a single `gas_caches` struct that contains all sub-structs. I worry that we're starting to litter the `Context` object (which is used in more places than just the gas estimation endpoints, IIUC).